src/host.cc

```diff
@@ -3358,9 +3336,8 @@ int host::process_check_result_3x(enum host::host_state new_state,
                                  int use_cached_result,
                                  unsigned long check_timestamp_horizon) {
   com::centreon::engine::host* master_host = NULL;
-  com::centreon::engine::host* temp_host = NULL;
-  objectlist* check_hostlist = NULL;
-  objectlist* hostlist_item = NULL;
+  host* temp_host;
+  std::list<host*> check_hostlist;
```

Comment:
you can perhaps switch to std::list<shared_ptr<host>> here

Reply:
This list is just some kind of reminder it is removed at the end of the function. And for several hosts, I don't have the shared_ptr container, only the host itself.